doc/ruleSets/PSR12.rst

```diff
@@ -11,7 +11,7 @@ Rules
 - `blank_line_after_opening_tag <./../rules/php_tag/blank_line_after_opening_tag.rst>`_
 - `braces <./../rules/basic/braces.rst>`_
   config:
-  ``['allow_single_line_anonymous_class_with_empty_body' => true]``
+  ``['allow_single_line_empty_body' => true]``
```

Comment:
empty bodies should be not a default in general, it decreases readability (for ex. on methods) and complicates diffing

Reply:
Not really. It does not force the rule, it just allows inlining them on purpose.

If we want to update rulesets to not use the newly deprecated option we are left with:
a) removing just the deprecated option, which is a BC break for sure.
b) replacing the deprecated option with the new one. That's not a BC break, since old code will continue to work as it worked.

Reply:
This option does not force empty bodies to be single line, it only *allows* that. I don't see that as a BC break.

Though this makes me think: since this codestyle is becoming more common, maybe we should propose an option to actually force that. So instead of `allow_single_line_empty_body => true|false`, maybe we could go with e.g. `empty_body => ignore|multi_line|single_line`?

Reply:
👍 , but for anonymous language constructs (anonymous function, class) there should be a separate configuration.

And for anonymous `single_line` should be allowed, for non-anonymous (class, ifaces, function, methods) `multi_line` should be enforced.

Reply:
Why the distinction between anonymous and non-anonymous?

Reply:
Because forcing `{}` to be on a single line for classes, methods etc. reduces readablity and complicates diffing.

Reply:
This applies to anonymous functions and classes.